src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/BotSegment.cs

```diff
@@ -346,25 +350,44 @@ private static BotSequenceEntry ParseSegment(string json, string fileName)
         {
             try
             {
+                var currentVersion = SdkApiVersion.CURRENT_VERSION;
                 var entry = new BotSequenceEntry
                 {
                     path = fileName,
                 };
 
+                // First we will try to parse the file as a SegmentList, if that fails we will try to parse
+                // the file as a standard Segment
                 try
                 {
-                    var segment = JsonConvert.DeserializeObject<BotSegmentList>(json);
-                    entry.description = segment.description;
-                    entry.entryName = segment.name;
+                    var segmentList = JsonConvert.DeserializeObject<BotSegmentList>(json);
+                    if (segmentList.EffectiveApiVersion >= currentVersion)
```

Comment:
This should be `>`

We are checking the bad case where the sequence/segment/etc is TOO new for the current runtime.